css/src/components/popover.scss

```diff
@@ -19,24 +21,52 @@ $popover-width: 224px !default;
 	.popover-content {
 		position: absolute;
 		width: $popover-width;
-		margin-block-start: $spacer-3;
 		padding: $spacer-5;
+		transform: none;
+		inset-inline-start: 0;
 		border: $popover-border;
 		border-radius: $popover-border-radius;
 		background-color: $popover-background-color;
 		box-shadow: $popover-shadow;
+
+		// hide popover until after position is calculated, then show
+		visibility: hidden;
```

Comment:
I don't think you need to tweak visibility of `popover-content`. `details` does it by default for you. Please double-check